src/uri.rs

```diff
@@ -1200,12 +1230,30 @@ impl PathAndQuery {
     pub fn from_shared(mut src: Bytes) -> Result<Self, InvalidUriBytes> {
         let mut query = NONE;
 
-        for i in 0..src.len() {
+        let mut i = 0;
+
+        while i < src.len() {
             let b = src[i];
 
             match URI_CHARS[b as usize] {
                 0 => {
-                    return Err(ErrorKind::InvalidUriChar.into());
+                    if b == b'%' {
```

Comment:
Could this just follow the pattern of `?` and `#` instead of matching 0 and then checking again what byte it is?

Reply:
`URI_CHARS` was also used in other situations where `%` isn't permitted. It seemed roughly equivalent to do it this way vs. create `PATH_CHARS` that permitted `%`.

Reply:
Oh, OK!